include/deal.II/lac/sparse_decomposition.templates.h

```diff
@@ -118,7 +118,8 @@ void SparseLUDecomposition<number>::initialize (
 
   // now use this sparsity pattern
   Assert (sparsity_pattern_to_use->n_rows()==sparsity_pattern_to_use->n_cols(),
-          typename SparsityPattern::ExcDiagonalNotOptimized());
+          ExcMessage ("It is not possible to compute this matrix decomposition for "
```

Comment:
It appears that the original exception was completely unrelated to the actual assert.